scripts/irods/test/test_ichmod.py

```diff
@@ -275,6 +275,36 @@ def test_ichmod_does_not_bypass_permission_model_for_collection__issue_6579(self
                     # Restore collection permissions or error will be thrown on cleanup
                     self.admin.assert_icommand(['ichmod', '-M', 'own', self.user.username, os.path.join(self.user.session_collection, collection)])
 
+    def test_ichmod_respects_strict_ACLs__issue_2570(self):
+        other_user = 'testuser'
+        self.admin.assert_icommand(['iadmin', 'mkuser', other_user, 'rodsuser'])
+
+        group = 'generalgroup'
+        self.admin.assert_icommand(['iadmin', 'mkgroup', group])
+        self.admin.assert_icommand(['iadmin', 'atg', group, self.user.username])
+
+        collections_to_test = ['specific1', 'specific2']
+        collection_base = os.path.join(self.admin.session_collection, 'general')
+        
+        self.admin.assert_icommand(['imkdir', '-p', os.path.join(collection_base, collections_to_test[0])])
+
+        self.admin.assert_icommand(['ichmod', 'modify_object', group, '-r', collection_base])
+        self.admin.assert_icommand(['ichmod', 'inherit', '-r', collection_base])
+
+        self.user.assert_icommand(['imkdir', os.path.join(collection_base, collections_to_test[1])])
+
+        for collection in collections_to_test:
+            with self.subTest(collection):
+                out, err, ec = self.user.run_icommand(['ichmod', 'read', other_user, os.path.join(collection_base, collection)])
+                self.assertEqual(ec, 8)
+                self.assertIn('CAT_NO_ACCESS_PERMISSION', err)
+                self.assertEqual(len(out), 0)
+
+        # Cleanup
+        self.admin.assert_icommand(['iadmin', 'rfg', group, self.user.username])
+        self.admin.assert_icommand(['iadmin', 'rmgroup', group])
+        self.admin.assert_icommand(['iadmin', 'rmuser', other_user])
```

Comment:
These should be in a `finally` block so that they are executed regardless of what happens.